src/components/List.jsx

```diff
@@ -0,0 +1,67 @@
+import { AiTwotoneDelete, AiTwotoneEdit } from "react-icons/ai";
+import { useContext } from "react";
+import { TiTick } from "react-icons/ti";
+import TodoContext from "../context/TodoContext";
+import { useState } from "react";
+
+const List = () => {
+  const { allTodos, handleDelete, todoCheck, setAllTodos } =
+    useContext(TodoContext);
+  const [isClicked, setIsClicked] = useState(false);
+  const [editInput, setEditInput] = useState("");
+
+  const handleEditInput = (event) => {
+    setEditInput(event.target.value);
+  };
+
+  const updateTodo = (i) => {
+    setAllTodos((prevtodos) => {
+      return prevtodos.map((item, index) => {
+        return index === i ? { ...item, text: editInput } : item;
+      });
+    });
+    setIsClicked(!isClicked);
+  };
+
+  const handleEdit = (i) => {
+    const text = allTodos.map((item, index) => {
+      if (index === i) return item.text;
+    });
+    setEditInput(text);
+    setIsClicked(!isClicked);
+  };
+
+  return (
+    <div>
+      {allTodos.map((todo, i) => {
+        return (
+          <div className="flex items-center justify-between w-[500px]">
```

Comment:
It's recommended to provide a map key for mapped JSX items. The map key could be like the id or index in this case.